packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -350,11 +355,48 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
                 Placeholder.configure({
                     placeholder: '',
                     showOnlyWhenEditable: false
+                }),
+                customLink.configure({
+                    // HTMLAttribute cannot be in camelCase as we want to match it with the name in Tiptap
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    HTMLAttributes: {
+                        rel: 'noopener noreferrer',
+                        target: null
+                    },
+                    autolink: true,
+                    openOnClick: false,
+                    linkOnPaste: false,
+                    validate: href => /^https?:\/\//i.test(href)
                 })
             ]
         });
     }
 
+    /**
+     * Extending the default link mark schema defined in the TipTap.
+     *
+     * "excludes": https://prosemirror.net/docs/ref/#model.MarkSpec.excludes
+     * "inclusive": https://prosemirror.net/docs/ref/#model.MarkSpec.inclusive
+     * "parseHTML": https://tiptap.dev/guide/custom-extensions#parse-html
+     * "renderHTML": https://tiptap.dev/guide/custom-extensions/#render-html
+     */
+    private getCustomLinkExtension(): Mark<LinkOptions> {
+        return Link.extend({
+            excludes: '_',
+            inclusive: false,
+            parseHTML() {
+                return [{ tag: anchorTag }];
```

Comment:
Is it correct that this is still using `nimble-anchor`? Or should it be symmetric with the `'a'` tag in `renderHTML`?

Reply:
`parseHTML` should include `nimble-anchor` because the markdown parser is shared between both the editor and the viewer. After setting the markdown, the parsed content will contain only `nimble-anchor` and during the rendering process in the editor, `renderHTML` will change the link node tag to `<a>` tag.

If we update `a` in `parseHTML` now, the editor will not identify that it has a link because the parsed node will not have any `a` but instead, it will have `nimble-anchor`. So link will render as plain text in the editor while parsing.

Another advantage of having this right now is, that if we change it to `nimble-anchor` in the future it is a simple one-line code change in `renderHTML` and we can avoid modifying the entire schema in the parser.

Reply:
Added a comment here